expo-boomboomapp/src/components/SongPicker.tsx

```diff
@@ -41,22 +41,25 @@ export function SongPicker({
   debounceTime = DEFAULT_DEBOUNCE_TIME,
 }: SongPickerProps) {
   const spotifyApiService = getGlobalInstance<SpotifyApiServiceI>(
-    ServiceInterface.SpotifyApiServiceI,
+    ServiceInterface.SpotifyApiServiceI
   );
   const languageService = getGlobalInstance<LanguageService>(
-    ServiceInterface.LanguageServiceI,
+    ServiceInterface.LanguageServiceI
   );
   const I18n = languageService.useTranslation();
   const onSearch = (searchString?: string): void => {
-    spotifyApiService
-      .fetchTracksNyName(searchString)
-      .then((tracks): void => {
-        setFetchedSongs(tracks);
-      })
-      .catch((error) => {
-        // TODO handle error better
-        console.log("spotifyApiService : ", error);
-      });
+    if (searchString != undefined && searchString != "") {
+      spotifyApiService
+        .fetchTracksNyName(searchString)
+        .then((tracks): void => {
+          setFetchedSongs([]);
+          setFetchedSongs(tracks.data);
+        })
+        .catch((error) => {
+          // TODO handle error better
+          console.log("spotifyApiService : ", error);
+        });
+    }
```

Comment:
Here you could use guard clause : ) 

```suggestion
    if (!searchString || searchString.trim() === "") {
       return; // No need to proceed if the search string is empty
    }
    
    spotifyApiService
      .fetchTracksNyName(searchString)
      .then((tracks) => {
         setFetchedSongs(tracks.data || []);
      })
      .catch((error) => {
          // TODO: handle error better
          console.error("spotifyApiService:", error);
      });
```

You can read more about this 'guard clause' [here](https://www.gavsblog.com/blog/how-to-and-why-use-guard-clauses-in-javascript). 
It's just a tiny suggestion, not a blocker to merge this PR !

Reply:
Yes it's a nice comment i will push changes